src/main/java/com/openshift/internal/restclient/model/template/Template.java

```diff
@@ -56,4 +56,14 @@ public Template(ModelNode node, IClient client, Map<String, String []> propertyK
 		return resources;
 	}
 
+	@Override
+	public void updateParameterValues(Collection<IParameter> parameters) {
+		Map<String, IParameter> actuals = getParameters();
+		for (IParameter param : parameters) {
+			if(actuals.containsKey(param.getName())) {
```

Comment:
not sure a boolean is enough. Do we provide a means to the user so that he can find out what values were not substituted? What about returning a collection of unknown keys? if it's empty we're all fine, if not it would contain the non-substitued keys (and we could warn the user about these)?